src/web/lib/components/ThemeColorsEditor/index.js

```diff
@@ -40,7 +41,14 @@ class ThemeColorsEditor extends React.Component {
   };
 
   handleColorChange = (name, color) => {
-    this.props.setColor({ name, color: color.rgb });
+    this.lastSelectedColor = color;
+    this.props.setColor({ name, color });
+  };
+
+  handleClearColor = (name) => {
+    const { theme: {colors} } = this.props;
+    this.lastSelectedColor = colors[name];
```

Comment:
Instead of assigning `lastSelectedColor` on change, assign it on render. This has several benefits:
- less code duplication.
- also works if props are changed by an external component.
- when switching between a color with a picker to a "default" color, the color picker will retain its color (instead of the color of the last click).

Something like this:

```diff
diff --git a/src/web/lib/components/ThemeColorsEditor/index.js b/src/web/lib/components/ThemeColorsEditor/index.js
index 2963e2f..6a19588 100644
--- a/src/web/lib/components/ThemeColorsEditor/index.js
+++ b/src/web/lib/components/ThemeColorsEditor/index.js
@@ -41,13 +41,10 @@ class ThemeColorsEditor extends React.Component {
   };
 
   handleColorChange = (name, color) => {
-    this.lastSelectedColor = color;
     this.props.setColor({ name, color });
   };
 
   handleClearColor = (name) => {
-    const { theme: {colors} } = this.props;
-    this.lastSelectedColor = colors[name];
     this.props.clearColor({name});
   };
 
@@ -86,6 +83,11 @@ class ThemeColorsEditor extends React.Component {
       )
     ];
 
+    const selectedColorValue = colors[selectedColor];
+    if (selectedColorValue) {
+      this.lastSelectedColor = selectedColorValue;
+    }
+
     return (
       <div className="theme-colors-editor">
         <div className="theme-colors-editor-main">
@@ -125,12 +127,12 @@ class ThemeColorsEditor extends React.Component {
           {selectedColor && advancedColors && 
             <div className="theme-colors-editor__options">
               <label>
-                <input type="radio" value="default" checked={!colors[selectedColor]}
+                <input type="radio" value="default" checked={!selectedColorValue}
                         onChange={ev => this.handleClearColor(selectedColor)}/>
                 use Firefox&#39;s default style for this color
               </label>
               <label>
-                <input type="radio" value="other" checked={!!colors[selectedColor]}
+                <input type="radio" value="other" checked={!!selectedColorValue}
                         onChange={ev =>
                           this.handleColorChange(selectedColor, this.lastSelectedColor)
                         }/>
@@ -140,8 +142,8 @@ class ThemeColorsEditor extends React.Component {
           }
           {selectedColor &&
             <SketchPicker
-              className={advancedColors && !colors[selectedColor] ? "theme-colors-editor__disabled" : ""}
-              color={colors[selectedColor] || this.lastSelectedColor}
+              className={advancedColors && !selectedColorValue ? "theme-colors-editor__disabled" : ""}
+              color={selectedColorValue || this.lastSelectedColor}
               width="270px"
               disableAlpha={!colorsWithAlpha.includes(selectedColor)}
               onChangeComplete={nextColor =>